### PR TITLE
feat(pages): update contract data mapping from difference sources

### PR DESCRIPTION
**Pull request by Poafs1**

### CODE REVIEW

File: src/lib/app-provider/env.ts [hunk 1]
```diff
@@ -45,6 +45,7 @@ export enum CELATONE_QUERY_KEYS {
   ADMINS_BY_CONTRACTS = "CELATONE_QUERY_ADMINS_BY_CONTRACTS",
   CONTRACT_MIGRATION_HISTORIES_BY_CONTRACT_ADDRESS = "CELATONE_QUERY_CONTRACT_MIGRATION_HISTORIES_BY_CONTRACT_ADDRESS",
   CONTRACT_MIGRATION_HISTORIES_BY_CONTRACT_ADDRESS_LCD = "CELATONE_QUERY_CONTRACT_MIGRATION_HISTORIES_BY_CONTRACT_ADDRESS_LCD",
+  CONTRACT_CW2_INFO = "CELATONE_QUERY_CONTRACT_CW2_INFO",
```
Comment (evilpeach) [inline src/lib/app-provider/env.ts:48]: CONTRACT_CW2_INFO_LCD

File: src/lib/services/wasm/contract/index.ts [hunk 2]
```diff
@@ -269,4 +272,17 @@ export const useInstantiatedContractsByAddressLcd = (
   );
 };
 
+export const useContractCw2Info = (
```
Comment (evilpeach) [inline src/lib/services/wasm/contract/index.ts:275]: add suffix `lcd`

File: src/lib/services/wasm/contract/lcd.ts [hunk 3]
```diff
@@ -150,3 +151,13 @@ export const getInstantiatedContractsByAddressLcd = (
...
       }));
     });
+
+export const getContractCw2InfoLcd = async (
+  endpoint: string,
+  contractAddress: BechAddr32
+) =>
+  axios
+    .get(
+      `${endpoint}/cosmwasm/wasm/v1/contract/${contractAddress}/raw/Y29udHJhY3RfaW5mbw%3D%3D`
```
Comment (evilpeach) [inline src/lib/services/wasm/contract/lcd.ts:161]: apply encodeURI

File: src/lib/services/types/wasm/contract.ts [hunk 4]
```diff
@@ -128,15 +128,14 @@ export const zContractLcd = zContractRest.transform<ContractData>((val) => ({
...
-    createdTimestamp: getDefaultDate(),
+    createdHeight: val.contract_info.created
+      ? Number(val.contract_info.created.block_height)
+      : null,
+    createdTimestamp: null,
     cw2Contract: null,
     cw2Version: null,
-    // TODO: make optional - get from histories
-    initMsg: "",
+    initMsg: "", // TODO: Get
```
Comment (songwongtp) [inline src/lib/services/types/wasm/contract.ts:138]: remove todo?

File: src/lib/types/contract.ts [hunk 5]
```diff
@@ -35,6 +35,7 @@ export interface ContractMigrationHistory {
   uploader: Nullable<BechAddr>;
   cw2Contract: Nullable<string>;
   cw2Version: Nullable<string>;
+  msg: string;
```
Comment (songwongtp) [inline src/lib/types/contract.ts:38]: `msg?: string`

File: src/lib/services/types/wasm/contract.ts [hunk 6]
```diff
@@ -168,7 +167,10 @@ const zMigrationHistoriesResponseItem = z
     timestamp: zUtcDate,
     uploader: zBechAddr,
   })
-  .transform<ContractMigrationHistory>(snakeToCamel);
+  .transform<ContractMigrationHistory>((val) => ({
+    ...snakeToCamel(val),
+    msg: "",
```
Comment (songwongtp) [inline src/lib/services/types/wasm/contract.ts:172]: revert

File: src/lib/pages/contract-details/data.ts [hunk 7]
```diff
@@ -1,16 +1,61 @@
...
+          contract: {
+            ...contractData.data.contract,
+            codeHash: code?.hash ?? "",
+            cw2Contract: cw2Info?.contract ?? null,
+            cw2Version: cw2Info?.version ?? null,
+            initMsg:
+              migrationHistories?.entries.find(
+                (history) =>
+                  history.remark?.operation ===
+                  "CONTRACT_CODE_HISTORY_OPERATION_TYPE_INIT"
```
Comment (songwongtp) [inline src/lib/pages/contract-details/data.ts:45]: use enum

File: src/lib/services/types/wasm/contract.ts [hunk 8]
```diff
@@ -191,17 +193,22 @@ export const zMigrationHistoriesResponseItemLcd = z
     operation: z.nativeEnum(RemarkOperation),
```
Comment (songwongtp) [inline src/lib/services/types/wasm/contract.ts:193]: let's create `zRemarkOperation` in `lib/types/contract.ts`, then use it in `zContractHistoryRemark` and here

**Review by songwongtp — CHANGES_REQUESTED** [review]

File: src/lib/services/wasm/contract/lcd.ts [hunk 9]
```diff
@@ -150,3 +151,13 @@ export const getInstantiatedContractsByAddressLcd = (
...
+
+export const getContractCw2InfoLcd = async (
+  endpoint: string,
+  contractAddress: BechAddr32
+) =>
+  axios
+    .get(
+      `${endpoint}/cosmwasm/wasm/v1/contract/${encodeURI(contractAddress)}/raw/Y29udHJhY3RfaW5mbw%3D%3D`
+    )
+    .then(({ data }) => JSON.parse(libDecode(data.data)) as ContractCw2Info);
```
Comment (evilpeach) [inline src/lib/services/wasm/contract/lcd.ts:163]: Maybe we should create Zod type for `ContractCw2Info`

Comment (Poafs1) [reply src/lib/services/wasm/contract/lcd.ts:163]: 555555 will check

**Review by songwongtp — APPROVED** [review]

**Review by evilpeach — APPROVED** [review]

**Pull request merged**